src/site/content/en/blog/image-support-for-async-clipboard/index.md

```diff
@@ -0,0 +1,317 @@
+authors:
+  - thomassteiner
+description: Chrome 76 adds expands the functionality of the Async Clipboard API to add support for png images. Copying and pasting images to the clipboard has never been easier.
+date: 2019-07-03
+updated: 2019-10-08
+tags:
+  - post # post is a required tag for the article to show up in the blog.
+  - capabilities
+  - chrome76
+  - cutandcopy
+  - execcommand
+  - input
+  - clipboard
+hero: hero.jpg
+alt: Clipboard with shopping list
+---
+
+Chrome 66, added support for the
+[text portion](https://developers.google.com/web/updates/2018/03/clipboardapi)
+of the [Asynchronous Clipboard API](https://w3c.github.io/clipboard-apis/#async-clipboard-api).
+Chrome 76 has added support for images makes it easy to programmatically
+copy and paste **png** images.
+
+{% Aside 'caution' %}
+  At the time of writing, only png files are supported. Support for other
+  images and file formats will be added in the future.
+{% endAside %}
+
+Before we dive in, let's take a brief look at how the Asynchronous Clipboard
+API works. If you remember the details, skip ahead to the [image section](#images).
+
+## Recap of the Asynchronous Clipboard API {: #recap }
+
+Before describing image support, I want to review how the Asynchronous Clipboard
+API works. Feel free to [skip ahead](#images) if you're already comfortable
+using the API.
+
+### Copy: writing text to the clipboard {: #copy-text }
+
+To copy text to the clipboard, call `navigator.clipboard.writeText()`.
+Since this API is asynchronous, the `writeText()` function returns a promise
+that resolves or rejects depending on whether the passed text is
+copied successfully. For example:
+
+```js
+async function copyPageURL() {
+  try {
+    await navigator.clipboard.writeText(location.href);
+    console.log('Page URL copied to clipboard');
+  } catch (err) {
+    console.error('Failed to copy: ', err);
+  }
+}
+```
+
+### Paste: reading text from the clipboard {: #reading-text }
+
+Much like copy, text can be read from the clipboard by calling
+`navigator.clipboard.readText()` and waiting for the returned promise to
+resolve with the text:
+
+```js
+async function getClipboardText() {
+  try {
+    const text = await navigator.clipboard.readText();
+    console.log('Clipboard contents: ', text);
+  } catch (err) {
+    console.error('Failed to read clipboard contents: ', err);
+  }
+}
+```
+
+### Handling paste events
+
+Paste events can be handled by listening for the (surprise) `paste` event.
+Note that you need to call `preventDefault()` in order to modify the
+to-be-pasted data, like for example, convert it to uppercase before pasting.
+It works nicely with the new asynchronous methods for reading clipboard text:
+
+```js
+document.addEventListener('paste', async (e) => {
+  e.preventDefault();
+  try {
+    const text = await navigator.clipboard.readText();
+    text = text.toUpperCase();
```

Comment:
Redefining a const?

Reply:
Easier to read than navigator.clipboard.readText().toUpperCase(). 

It's also pretty typical. Massaging strings in steps is something you often do in text manipulation contexts.

Reply:
Fixed.